### PR TITLE
William/cleanups

### PR DESCRIPTION
**Pull request by swansontec**
Working on the `asValue` improvements revealed a few other glitches in the repo.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -89,7 +89,7 @@ Since JSON doesn't have its own date type, people usually send dates as strings:
 { "birthday": "2010-04-01" }
 ```
 
-It's not enough to check that `birthday` is a string - the contents need to be parsed and validated as well. Fortunately, cleaners can do this. The `asDate` cleaner will actually parse strings into Javascript date objects, solving this problem.
+It's not enough to check that `birthday` is a string - the contents need to be parsed and validated as well. Fortunately, cleaners can do this. The `asDate` cleaner will actually parse strings into Javascript `Date` objects, solving this problem.
```
Comment (samholmes) [inline docs/README.md]: While we're here: `s/Javascript/JavaScript`

File: test/cleaners/primitives.test.js [hunk 2]
```diff
@@ -26,7 +24,7 @@ describe('asNumber', function () {
   it('accepts only numbers', function () {
     expect(asNumber(0)).equals(0)
     expect(asNumber(-1)).equals(-1)
-    expect(asNumber(NaN)).is.NaN
+    expect(asNumber(NaN)).deep.equals(NaN)
```
Comment (samholmes) [inline test/cleaners/primitives.test.js:27]: `deep.equals`?

File: package.json [hunk 3]
```diff
@@ -35,7 +35,7 @@
     "lint": "eslint .",
     "precommit": "lint-staged && npm-run-all -p flow types",
     "prepare": "husky install && npm-run-all clean -p lib types",
-    "test": "nyc mocha 'test/**/*.test.js'",
+    "test": "nyc mocha 'test/**/*.test.ts'",
```
Comment (samholmes) [inline package.json:38]: Does mocha support TS without configuration now, or is it configured with sucrase or tsc already?

**Review by samholmes — CHANGES_REQUESTED** [review]

File: test/cleaners/primitives.test.js (hunk 2 above)
Comment (swansontec) [reply test/cleaners/primitives.test.js:27]: Regular `.equals` uses `===`, which doesn't work with `NaN`. The `deep.equals` uses a much fancier algorithm that does handle `NaN` correctly.

File: package.json (hunk 3 above)
Comment (swansontec) [reply package.json:38]: The `.nycrc` file has the `sucrase/register` settings, so `nyc mocha` handles TypeScript without trouble.

File: test/cleaners/primitives.test.js (hunk 2 above)
Comment (samholmes) [reply test/cleaners/primitives.test.js:27]: Gotta love DSLs. Thanks for the explainer.

**Review by samholmes — APPROVED** [review]

**Pull request merged**